TINF-163: Favorites hinzugefügt

Leaqua07 opened the pull request:
-Endpunkte hinzugefügt:
1. '/users/self/favorits/{id}':
- post und delete
2. '/users/self/favorits/':
- get

Leaqua07 committed:
Favorits hinzugefügt:
-Endpunkte hinzugefügt:
1. '/users/self/favorits/{id}':
- post und delete
2. '/users/self/favorits/':
- get

jstsddg commented on api/v2/openapi.yaml:
```diff
@@ -115,6 +115,70 @@ paths:
+          name: id
+          description: id of the location
+          required: true
+      requestBody:
+        description: id of the location
+        required: true
+        content:
+          application/json:
+            schema:
+              $ref: '#/components/schemas/Location'
```
Brauchen wir überhaupt einen `requestBody`? Die ID steht ja schon in der URL

jstsddg commented on api/v2/openapi.yaml:
```diff
@@ -115,6 +115,70 @@ paths:
+                type: array
+                items:
+                  $ref: '#/components/schemas/Location'
+        '400':
+          description: Malformed request
```
Hier könnte zusätzlich noch ein `404` auftreten, wenn die Location nicht existiert und ein `401`, wenn der man nicht eingeloggt ist.

jstsddg commented on api/v2/openapi.yaml:
```diff
@@ -115,6 +115,70 @@ paths:
+          name: id
+          description: id of the location
+          required: true
+      requestBody:
+        description: id of the location
+        required: true
+        content:
+          application/json:
+            schema:
+              $ref: '#/components/schemas/Location'
```
Hier brauchen wir glaube ich ebenso keinen `requestBody`.

jstsddg commented on api/v2/openapi.yaml:
```diff
@@ -115,6 +115,70 @@ paths:
+      responses:
+        '200':
+          description: Success
+        '400':
+          description: Malformed request
```
Hier könnte auch ein `401`, bzw. `404` auftreten.

jstsddg commented on api/v2/openapi.yaml:
```diff
@@ -115,6 +115,70 @@ paths:
+                type: array
+                items:
+                  $ref: '#/components/schemas/Location'
+        '404':
+          description: Not found
```
Ich glaube hier kann kein `404` auftreten, da wir wenn man keine Favoriten hat einfach eine leere Liste `[]` zurück geben.

jstsddg replied:
Aber auch hier könnte wieder ein `401` auftreten.

jstsddg reviewed (changes requested):
Statt `favorits` heißt es `favorites`, das habe ich vorhin glaube ich falsch gesagt. Ich fixe das gerade kurz.

jstsddg committed:
Fixed 'favorites' spelling

jstsddg committed:
Fixed missing schema in parameter

jstsddg committed:
Rearanged endpoints for better readability

jstsddg commented on api/v2/openapi.yaml:
```diff
@@ -116,6 +116,76 @@ paths:
+                type: array
+                items:
+                  $ref: '#/components/schemas/Location'
+        '404':
+          description: Not found
```
Ich glaube hier kann kein `404` auftreten, da wir wenn man keine Favoriten hat einfach eine leere Liste `[]` zurück geben.

Aber hier könnte auch wieder ein `401` auftreten, wenn der Benutzer nicht angemeldet ist.

MrKraboom commented:
Ich habe den Tippfehler entfernt, die Endpunkt Reihenfolge angepasst (get, post, delete) und ein fehlendes Schema hinzugefügt.

MrKraboom commented:
Die `Location`s sollten noch ein `favorite`-Flag bekommen ob die Location ein Favorit ist. Möglicherweise `null` wenn man nicht eingeloggt ist?

Leaqua07 committed:
Favorits hinzugefügt:
-Endpunkte angepasst:
GET:
- 404 entfernt
POST:
- Requestbode removed
- 404 und 401 added
DELETE:
- Requestbode removed
- 404 und 401 added

jstsddg committed:
Fixed error messages

jstsddg committed:
Added favorite flag to Location

jstsddg committed:
Fixed bad example

jstsddg reviewed (dismissed):
LGTM

MrKraboom commented:
@SAKPaaS/frontend Gerne auch ein Review vom Frontend, damit wir hier nicht unbrauchbares implementieren.

nclskfm commented on api/v2/openapi.yaml:
```diff
@@ -116,6 +116,68 @@ paths:
+      description: List all favorite locations for the currently logged in user.
+      tags:
+        - "User: Favorites"
+      responses:
```
kann nicht hier auch ein `401` auftreten? Wenn der User nicht eingeloggt ist? Oder was passiert dann?

jstsddg replied:
Ja stimmt, @Leaqua07 kannst du das gerade kurz hinzufügen oder soll ich das nachher machen?

jstsddg replied:
Ich hab's jetzt gerade einfach kurz eingefügt-

jstsddg committed:
Added 401 to /users/self/favorites

jstsddg reviewed (approved):
401 Implementiert, sieht gut aus
